### PR TITLE
feat(cli): Add better unit testing.

### PR DESCRIPTION
**Pull request by c-r33d**
## Summary                                   
                                                                                                                                                                                                                                                                                         
  Tighten unit-test coverage and simplify design across the namespaced-policy migration planner. Package coverage 78.2% → 80.9%, with targeted behavioral gaps closed in the planner pipeline.                                                                                           
  
  ## Design changes                                                                                                                                                                                                                                                                      
                  
  - **Removed dead code:** `ActionReference`/`ActionReferenceKind` accumulator and `ErrDuplicateCanonicalMatch` sentinel (never surfaced to users or callers).                                                                                                                           
  - **First-match resolution:** the 5 `resolveExisting*` matchers now return the first canonical match instead of erroring on duplicates, keeping resolver behavior consistent across all target types.
  - **Plan-time dependency validation:** registered resources and obligation triggers now validate their action dependencies during resolve (matching subject mappings), rather than failing later at execute time.                                                                      
  - **Typed sentinel errors** for dependency failures (`ErrMissingActionID`, `ErrUnresolvedActionDependency`, etc.) so tests can assert via `errors.Is` while preserving contextual wrapping.                                                                                            
  - **Consistent nil-candidate handling:** all 5 candidate types in `deriveTargets` now skip nil entries instead of mixing error/skip behavior.                                                                                                                                          
  - Documented the interactive-path vs initial-pass convergence on `ensureRegisteredResourceActionResolution`.                                                                                                                                                                           
                                                                                                                                                                                                                                                                                         
  ## Test additions                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                         
  - `canonical.go`: table-driven tests for all 5 `*CanonicalEqual` wrappers covering nil, cosmetic-equivalence (case/whitespace/order), distinguishing cases, and empty-field gating.                                                                                                    
  - `resolved.go`: first-match-wins tests; typed-error assertions for all dependency validation paths; scope-gating drops; already-migrated routing for SCS/SM/OT.
  - `derived.go`: nil-skip tests for all 5 candidate types; namespace-accumulator dedupe; FQN-fallback resolution.                                                                                                                                                                       
  - `interactive_commit.go`: abort propagation test (reviewer returns `ErrInteractiveReviewAborted` and stops prompting).                                                                                                                                                                
  - `plan.go`: `LookupActionTarget` and `LookupSubjectConditionSetTarget` table-driven tests (nil, empty inputs, match, not-found).                                                                                                                                                      
  - `summary.go`: `recordTargetStatus` table-driven test covering all 4 status branches plus accumulation.                                                                                                                                                                               
                                                                                                                                                                                                                                                                                         
  ## Coverage deltas                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                         
  | File | Before | After |                                                                                                                                                                                                                                                              
  |---|---|---|   
  | `canonical.go` | ~85% | 95.7% |
  | `resolved.go` | ~88% | 94.4% |                                                                                                                                                                                                                                                       
  | `plan.go` | 68.9% | 84.2% |
  | `summary.go` | 85.1% | 86.5% |                                                                                                                                                                                                                                                       
  | **Package total** | **78.2%** | **80.9%** | 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * More robust handling of missing, nil, or ambiguous dependencies during policy migration and resolution; clearer, specific error outcomes and stable nil-handling to avoid partial/misleading results.

* **Refactor**
  * Simplified plan and resolution model: removed namespace-level aggregation and propagated reference metadata, streamlined target determination and interactive review flows.

* **Tests**
  * Expanded unit and e2e coverage for canonical equality, derivation, resolution, interactive aborts, cross-namespace migration, and many edge-case nil/missing scenarios.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by c-r33d** [comment]
@coderabbitai review

File: otdfctl/migrations/namespacedpolicy/resolved.go [hunk 1]
```diff
@@ -427,116 +397,93 @@ func (r *resolver) resolveSubjectMappingDependencies(mapping *policy.SubjectMapp
 	for _, action := range mapping.GetActions() {
 		actionID := action.GetId()
 		if actionID == "" {
-			return errors.New("subject mapping references an action without an id")
+			return ErrMissingActionID
 		}
```
[the remark this answers is not in the document]
Comment (c-r33d) [reply otdfctl/migrations/namespacedpolicy/resolved.go:401]: the protobuf getters are nil safe.

**Review by elizabethhealy — APPROVED** [review]
